primary_beam_correction.py

```diff
@@ -207,63 +195,88 @@ def write_new_fits(pbc_image, path, outputFilename):
     hdr = images.header
     newhdr = hdr.copy()
     # add in required beam keywords
-    if newhdr['CLEANBMJ'] < 0:
-        # add in required beam keywords
-        newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
-        newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
-        newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
+    try:
+        if newhdr['CLEANBMJ'] < 0:
+            # add in required beam keywords
+            newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
+            newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
+            newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
 
-        newhdr['CTYPE3'] = 'FREQ'
-        newhdr['NAXIS3'] = 1
-        newhdr['CDELT3'] = 1.0
-        newhdr['CRVAL3'] = 1.284e9
-    else:
-        # add in required beam keywords
-        newhdr['BMAJ'] = newhdr['CLEANBMJ']
-        newhdr['BMIN'] = newhdr['CLEANBMN']
-        newhdr['BPA'] = newhdr['CLEANBPA']
-        # change the frequency plane keywords, we don't want multiple frequency
-        newhdr['CTYPE3'] = 'FREQ'
-        newhdr['NAXIS3'] = 1
-        newhdr['CDELT3'] = 1.0
+            newhdr['CTYPE3'] = 'FREQ'
+            newhdr['NAXIS3'] = 1
+            newhdr['CDELT3'] = 1.0
+            newhdr['CRVAL3'] = 1.284e9
+        else:
+            # add in required beam keywords
+            newhdr['BMAJ'] = newhdr['CLEANBMJ']
+            newhdr['BMIN'] = newhdr['CLEANBMN']
+            newhdr['BPA'] = newhdr['CLEANBPA']
+            # change the frequency plane keywords, we don't want multiple frequency
+            newhdr['CTYPE3'] = 'FREQ'
+            newhdr['NAXIS3'] = 1
+            newhdr['CDELT3'] = 1.0
+    except Exception:
+        logging.error('Exception occurred, keywords not found', exc_info=True)
 
     data_new = pbc_image[:, 0:1, :, :]  # the shape of data_new = (1,1, npix, npix)
     new_hdu = fits.PrimaryHDU(header=newhdr, data=data_new)
-    new_hdu.writeto(outputFilename)
+    new_hdu.writeto(outputFilename, overwrite=True)
     return
 
 
-if __name__ == "__main__":
-    args = docopt(__doc__)
-    path = args['--path']
-    # Reading data
-    print('-------------------------')
-    print('Reading in the fits file')
+def config_logging(message):
+    logging.basicConfig(format='%(message)s', level=logging.INFO)
+    return logging.info(message)
+
+
+def create_parser():
+    parser = argparse.ArgumentParser("Input an MeerKAT SDP pipeline continuum image and prouduce "
+                                     "primary beam corrected image in a direcectory same as that "
+                                     "of an input image.")
+    requiredNamed = parser.add_argument_group('required named arguments')
```

Comment:
Variable names should not contain capitals, only classes.

See: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Reply:
fixed